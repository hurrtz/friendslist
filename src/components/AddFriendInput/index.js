import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import styles from './styles.css';

class AddFriendInput extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      name: props.name || '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ name: e.target.value });
  }

  handleSubmit(e) {
    const { addFriend } = this.props;
    const name = e.target.value.trim();

    if (e.which === 13) {
      addFriend(name);
      this.setState({ name: '' });
    }
  }

  render() {
    const { name } = this.state;

    return (
      <input
        type="text"
        autoFocus="true"
        className={classnames('form-control', styles.addFriendInput)}
        placeholder="Type the name of a friend"
        value={name}
        onChange={this.handleChange}
        onKeyDown={this.handleSubmit}
      />
    );
  }
}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired,
};

export default AddFriendInput;
