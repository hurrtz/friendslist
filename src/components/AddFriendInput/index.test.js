import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import AddFriendInput from './';

Enzyme.configure({ adapter: new Adapter() });

function setup(injectedProps = {}) {
  const props = {
    addFriend: jest.fn(),
    ...injectedProps,
  };

  const enzymeWrapper = shallow(<AddFriendInput {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('AddFriendInput', () => {
  it('should render self and subcomponents', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.find('input').exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<AddFriendInput addFriend={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
