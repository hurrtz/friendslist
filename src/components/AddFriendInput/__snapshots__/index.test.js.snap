exports[`AddFriendInput should match the snapshot 1`] = `
<input
  autoFocus="true"
  className="form-control"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Type the name of a friend"
  type="text"
  value="" />
`;
